### PR TITLE
fix: validation working for multiple tickets for all fields(name , price, quantity etc)

### LINKED ISSUE #3797: Required field quantity of tickets not checked in the create event page if more than one ticket category is involved
**Describe the bug**
During event creation if you have more than one category of tickets (eg. 'economy' and 'reserved') of any type(free or paid). Only the quantity of the first ticket is validated others are not.

**To Reproduce**
1. Login and go to the create event page.
2. Fill all the required details except ticket details.
2. Add a type(paid or free) of ticket(the validity of this ticket is checked).
3. Add another type(paid or free) of ticket and don't fill in the quantity of this one.
4. Publish the event.
5. The event is published successfully.

**Expected behavior**
The page should prompt the user to enter the quantity.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 79.0.3945.117 (Official Build) (64-bit)

i want to work on this issue.

### PR DESCRIPTION
**Pull request by abhiandthetruth**
Fixes #3797

#### Short description of what this resolves:
During event creation if you have more than one category of tickets of any type(free or paid). Only the quantity of the first ticket is validated others are not. This pull requests fixes this issue and enables validation for all tickets. 

#### Changes proposed in this pull request:
1. In app\templates\components\forms\wizard\basic-details-step.hbs
 -> ticket index is passed as a parameter to the ticket-input widget for each ticket created.

2. In app\templates\components\widgets\forms\ticket-input.hbs
 -> All the inputs to be validated have additional 'data-validate' attribute which is it's name concatenated with the current ticket's index. eg: 'ticket_quantity_0' for the first ticket row.

3. In app\components\forms\wizard\basic-details-step.js
 -> All the ticket field validation rules are now generated from the function 'ticketsValidatonRules' which is called by the 'getValidationRules' function at the end. The implemented function returns rules for all the fields with identifiers same as their 'data-validate' attribute. The identifiers are calculated in a loop. Thus whenever a new ticket is added, new rules specific to it are added.

After the fix:
Untitled

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by abhiandthetruth** [comment]
@kushthedude  @uds5501 please review

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -91,6 +91,121 @@ export default Component.extend(FormMixin, EventWizardMixin, {
     return validationRules;
   }),
 
+  ticketsValidationRules() {
```
Comment (kushthedude) [inline app/components/forms/wizard/basic-details-step.js]: For what good is frontend validation when the server is still accepting such data. 
There should be server checks for such empty fields, Also I am not sure that we need to roll back to native jquery validation. 
I will wait for other mentors to review on this one.
cc: @iamareebjamal @uds5501 @mrsaicharan1

**Comment by iamareebjamal** [comment]
Yup. Needs validation in server

**Comment by abhiandthetruth** [comment]
Please consider this issue in the semantic-ui github repo before reviewing
https://github.com/Semantic-Org/Semantic-UI/issues/2688
which exposes it's limitation in these cases. I think this is the only work around. I may be wrong though

**Comment by iamareebjamal** [comment]
This is not related to frontend, so linking to semantic UI issue makes no sense. Server should not accept invalid values, period. When the server bug is solved, we can move with frontend `enhancement` and merge this

**Comment by kushthedude** [comment]
Stale!

**Pull request closed without merging**